library/src/main/scala/scala/scalajs/js/Symbol.scala

```diff
@@ -0,0 +1,41 @@
+/*                     __                                               *\
+**     ________ ___   / /  ___      __ ____  Scala.js API               **
+**    / __/ __// _ | / /  / _ | __ / // __/  (c) 2013-2017, LAMP/EPFL   **
+**  __\ \/ /__/ __ |/ /__/ __ |/_// /_\ \    http://scala-js.org/       **
+** /____/\___/_/ |_/____/_/ | |__/ /____/                               **
+**                          |/____/                                     **
+\*                                                                      */
+
+
+package scala.scalajs.js
+
+import scala.scalajs.js
+import scala.scalajs.js.annotation._
+
+/** <span class="badge badge-ecma6" style="float: right;">ECMAScript 6</span>
+ *  JavaScript Symbol.
+ */
+@js.native
+sealed trait Symbol extends js.Any
+
+object Symbol {
+  @JSName("Symbol")
+  @js.native
+  private object NativeSymbol extends js.Object {
+    def apply(): Symbol = js.native
+    def apply(description: String): Symbol = js.native
+
+    def `for`(key: String): Symbol = js.native
+    def keyFor(sym: Symbol): js.UndefOr[String] = js.native
+
+    val iterator: Symbol = js.native
+  }
```

Comment:
Because at some point we realized we would have want `js.Array` to be non-native, to include custom things in there. I'm mostly anticipating the need to evolve `Symbol` into more than just its facade.

Also, for 1.0 I think we should turn the type `js.Symbol` into a hijacked class. We have hijacked classes to represent all other JS primitive types, but not `symbol`. I think it would make sense. In that case, it would also be weird, possibly problematic, for the companion object to be a native.

But I don't have very strong feelings about this.